File: payroll-api/src/main/java/com/company/payroll/mapper/StaffLeaveMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffLeave;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffLeaveMapper {
    int deleteByPrimaryKey(Integer lId);
    int insert(StaffLeave row);
    int insertSelective(StaffLeave row);
    List<StaffLeave> selectList();
    List<StaffLeave> selectListByStaffId(Integer staffId);
    StaffLeave selectByPrimaryKey(Integer lId);
    int updateByPrimaryKeySelective(StaffLeave row);
    int updateByPrimaryKey(StaffLeave row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffLeaveMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffLeave
import org.springframework.stereotype.Repository


@Repository
interface StaffLeaveMapper {
  fun deleteByPrimaryKey(lId: Int): Int

  fun insert(row: StaffLeave): Int

  fun insertSelective(row: StaffLeave): Int

  fun selectByPrimaryKey(lId: Int): StaffLeave

  fun selectList(): List<StaffLeave>

  fun selectListByStaffId(staffId: Int): List<StaffLeave>

  fun updateByPrimaryKeySelective(row: StaffLeave): Int

  fun updateByPrimaryKey(row: StaffLeave): Int
}